setup

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# -e (-o errexit): exit immediately on error
+# -u (-o nounset): treat unset variables as erros
+# -o pipefail: propagate intermediate errors in pipelines
+set -euo pipefail
+
+main() {
+  print_step "Verifying that pip is available"
+  pip=$(command -v pip3 || command -v pip || echo -n "")
+  if [ -z "$pip" ]; then
+    print_error "Cannot find pip, aborting."
+    exit 1
+  fi
+
+  print_step "Installing and initializing pipenv"
+  $pip install pipenv
+
+  print_step "Installing Python dependencies"
+  pipenv install
+
+  print_step "Generating Ansible Vault secret"
+  pwgen -1 24|tee ~/.forem_selfhost_ansible_vault_password
```

Comment:
This needs to be idempotent so it doesn't clobber the vault password. If they encrypt their secrets with one password and then run the script again, it will nuke their old one and make life difficult. 

```
# pwgen -1 24|tee ~/.forem_selfhost_ansible_test
foorai2oshuB9looWi9leegh
# cat ~/.forem_selfhost_ansible_test
foorai2oshuB9looWi9leegh
# pwgen -1 24|tee ~/.forem_selfhost_ansible_test
eeVeiZae8ieya7tookohphei
# cat ~/.forem_selfhost_ansible_test 
eeVeiZae8ieya7tookohphei
```

Reply:
Idempotency for the various steps was one of my open questions, so 👍

Reply:
@jdoss Right, I should have answered late at night as I did last night because my answer was terse and didn't really help much. The reason idempotency was an open question for me is that this script would probably only be run once, for the very first provisioning of the Forem. After that the necessary libraries are installed, the secretes generated etc. and users can just make their modifications in a text editor and run ansible, they shouldn't need this script anymore.

I was thinking about doing this via some sentinel file, i.e. if a specific empty file exists, the script outputs a message and terminates. Though even given that we want to make sure we don't accidentally clobber user's data and make life harder for them, so I'll definitely add more idempotency checks.

Reply:
All good! I think being able to safely run the script to protect the inventory and most importantly the Ansible vault secret file are def worth the effort here.

Reply:
Yes, agreed and already added.